frontend/src/metabase/models/components/ModelDetailPage/ModelDetailHeader/ModelDetailHeader.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useCallback, useMemo, useState } from "react";
+import { t } from "ttag";
+
+import Button from "metabase/core/components/Button";
+import EntityMenu from "metabase/components/EntityMenu";
+import Link from "metabase/core/components/Link";
+import Modal from "metabase/components/Modal";
+
+import * as Urls from "metabase/lib/urls";
+
+import ArchiveModelModal from "metabase/questions/containers/ArchiveQuestionModal";
+import CollectionMoveModal from "metabase/containers/CollectionMoveModal";
+
+import type { Collection } from "metabase-types/api";
+
+import type Question from "metabase-lib/Question";
+
+import {
+  ModelHeader,
+  ModelHeaderButtonsContainer,
+  ModelTitle,
+  ModelFootnote,
+} from "./ModelDetailHeader.styled";
+
+interface Props {
+  model: Question;
+  onChangeName: (name?: string) => void;
+  onChangeCollection: (collection: Collection) => void;
+}
+
+type HeaderModal = "move" | "archive";
+
+function ModelDetailHeader({ model, onChangeName, onChangeCollection }: Props) {
+  const [modal, setModal] = useState<HeaderModal | null>(null);
+
+  const modelCard = model.card();
+  const canWrite = model.canWrite();
+
+  const queryEditorLink = Urls.modelEditor(modelCard, { type: "query" });
+  const exploreDataLink = Urls.model(modelCard);
+
+  const extraActions = useMemo(() => {
+    return [
+      {
+        title: t`Move`,
+        icon: "move",
+        action: () => setModal("move"),
+      },
+      {
+        title: t`Archive`,
+        icon: "archive",
+        action: () => setModal("archive"),
+      },
+    ];
+  }, []);
+
+  const handleCloseModal = useCallback(() => setModal(null), []);
+
+  const handleCollectionChange = useCallback(
+    (collection: Collection) => {
+      onChangeCollection(collection);
+      handleCloseModal();
+    },
+    [onChangeCollection, handleCloseModal],
+  );
+
+  const renderModal = useCallback(() => {
```

Comment:
I'm not sure I can see the problem with calling it instantly. `useCallback` should make sure we don't instantiate a new function on every component render as it needs variables that don't change that often

Reply:
There's no problem with it being called instantly per se. But it could be seen as a distraction IMO since my mental model is that every `useCallback` or `useMemo` should be there for a reason, [whether for referential comparison or better performance.](https://kentcdodds.com/blog/usememo-and-usecallback#so-when-should-i-usememo-and-usecallback)

again this is really minor.

Ref: https://kentcdodds.com/blog/usememo-and-usecallback#why-is-usecallback-worse